service/worker/diagnostics/invariants/invariant.go

```diff
@@ -28,13 +28,13 @@ import "context"
 type InvariantCheckResult struct {
 	InvariantType string
 	Reason        string
-	Metadata      []byte
+	Metadata      any
```

Comment:
This will lose all type information upon traversing from activity to workflow or workflow to activity.

JSON will parse objects to a map[string], any of which you won't be able to type assert by a specific type.